numpy/core/src/multiarray/mapping.c

```diff
@@ -727,8 +727,11 @@ prepare_index(PyArrayObject *self, PyObject *index,
         }
     }
     else if (used_ndim > PyArray_NDIM(self)) {
-        PyErr_SetString(PyExc_IndexError,
-                        "too many indices for array");
+        PyErr_Format(PyExc_IndexError,
+                     "array has %"NPY_INTP_FMT
+                     "-dimension, but %d were indexed",
+                     PyArray_NDIM(self),
+                     used_ndim);
```

Comment:
It would be good to keep `"too many"` in the message somewhere.

Reply:
Could you give some example code?

Reply:
Note that `ndim` is an integer IIRC, so it should just use `%d`. I am not sure what you mean with example code? Eric is merely suggesting to start the message with something like the old message (maybe rearranged) before you go into the details.

Reply:
I see! Sorry, I misunderstood that.

Reply:
How about `too many indices for array: array has 2-dimension, but 5 were indexed`? Is it more clear?